libpoolprotocols/PoolURI.cpp

```diff
@@ -0,0 +1,98 @@
+#include <libpoolprotocols/PoolURI.h>
+#include <map>
+
+using namespace dev;
+using namespace std;
+
+static void toLower(string &str)
+{
+	string::iterator end = str.end();
+	for (string::iterator it = str.begin(); it != end; it++)
+		*it = tolower(*it);
+}
+
+typedef struct {
+	ProtocolFamily family;
+	SecureLevel secure;
+	unsigned version;
+} SchemeType;
+
+static map<string, SchemeType> s_schemes = {
+	{"stratum",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  0}},
+	{"stratum1",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  1}},
+	{"stratum2",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  2}},
+	{"stratum+tcp",	{ProtocolFamily::STRATUM, SecureLevel::NONE,  0}},
```

Comment:
hm, this suggests, that stratum normal is not tcp.  But in fact, we only support stratum tcp. (no over http or websocket)

Reply:
For me it suggested stratum over default transport layer, but you may be right.

Reply:
it actually does use http  :) (jsonrpc::HttpClient)